pkg/autoscaler/metrics/collector_test.go

```diff
@@ -208,6 +208,72 @@ func TestMetricCollectorScraper(t *testing.T) {
 	}
 }
 
+func TestMetricCollectorNoScraper(t *testing.T) {
+	logger := TestLogger(t)
+
+	mtp := &fake.ManualTickProvider{
+		Channel: make(chan time.Time),
+	}
+	now := time.Now()
+	metricKey := types.NamespacedName{Namespace: defaultNamespace, Name: defaultName}
+	const (
+		reportConcurrency = 0.0
+		reportRPS         = 0.0
+		wantConcurrency   = 0.0
+		wantRPS           = 0.0
+		wantPConcurrency  = 0.0
+		wantPRPS          = 0.0
+	)
+	stat := Stat{
+		Time:                      now,
+		PodName:                   "testPod",
+		AverageConcurrentRequests: reportConcurrency,
+		RequestCount:              reportRPS,
+	}
+	scraper := &testScraper{
+		s: func() (Stat, error) {
+			return stat, nil
+		},
+	}
+	factory := scraperFactory(scraper, nil)
+
+	coll := NewMetricCollector(factory, logger)
+	coll.tickProvider = mtp.NewTicker // custom ticker.
+
+	noTargetMetric := defaultMetric
+	noTargetMetric.Spec.ScrapeTarget = ""
+	coll.CreateOrUpdate(&noTargetMetric)
+
+	// Tick three times.  Time doesn't matter since we use the time on the Stat.
+	mtp.Channel <- now
+	mtp.Channel <- now
+	mtp.Channel <- now
+
+	wait.PollImmediate(10*time.Millisecond, 100*time.Millisecond, func() (bool, error) {
+		_, _, errCon := coll.StableAndPanicConcurrency(metricKey, now)
+		_, _, errRPS := coll.StableAndPanicRPS(metricKey, now)
+		return errRPS == ErrNoData && errCon == errRPS, nil
+	})
+
+	gotConcurrency, panicConcurrency, _ := coll.StableAndPanicConcurrency(metricKey, now)
+	gotRPS, panicRPS, noData := coll.StableAndPanicRPS(metricKey, now)
+	if noData != ErrNoData {
+		t.Errorf("StableAndPanicRPS = %v", noData)
+	}
+	if panicConcurrency != wantPConcurrency {
+		t.Errorf("PanicConcurrency() = %v, want %v", panicConcurrency, wantPConcurrency)
+	}
+	if panicRPS != wantPRPS {
+		t.Errorf("PanicRPS() = %v, want %v", panicRPS, wantPRPS)
+	}
+	if gotConcurrency != wantConcurrency {
+		t.Errorf("StableConcurrency() = %v, want %v", gotConcurrency, wantConcurrency)
+	}
+	if gotRPS != wantRPS {
+		t.Errorf("StableRPS() = %v, want %v", gotRPS, wantRPS)
+	}
```

Comment:
Can we add a case where you manually call `Record` on the collector, and we get the appropriate values back? Just to enforce that that path (the activator path essentially) continues to work.

Reply:
Fixed in 7b1feced1c20f12f662502628344d8087c3978f6.

Let me know if the updated code coverage is sufficient.